icechunk/src/session.rs

```diff
@@ -1376,14 +1376,40 @@ impl<'a> FlushProcess<'a> {
     ) -> SessionResult<()> {
         let mut from = vec![];
         let mut to = vec![];
+
+        // iterator of chunks for a single node
         let chunks = stream::iter(
             self.change_set
                 .new_array_chunk_iterator(node_id, node_path)
                 .map(Ok::<ChunkInfo, Infallible>),
         );
+
+        let shards = ManifestShards::from_edges(vec![vec![0, 10, 20], vec![0, 10, 20]]);
+
+        // for it in chunks {}
+
+        // if manifest-split-size is set then we know extents, this is then duplicated between snapshot and config
+        //  1. Should we record the "real" extent or the theoretical-max extent (as set in the config).
+        //  2.
+        // if not, we aggregate_extents
+        // this is a choice, we could just always aggregate extents
         let chunks = aggregate_extents(&mut from, &mut to, chunks, |ci| &ci.coord);
```

Comment:
one q here: Now that the extents are defined by the shard sizes, then this function is not needed? Unless we want to record "true" extents for some reason.

Reply:
I think actual extents is useful...